plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java

```diff
@@ -666,10 +665,7 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
                         RedshiftClient::writeChar));
 
             case Types.VARCHAR: {
-                if (type.columnSize().isEmpty()) {
-                    throw new TrinoException(REDSHIFT_INVALID_TYPE, "column size not present");
-                }
-                int length = type.requiredColumnSize();
+                int length = type.columnSize().orElse(REDSHIFT_MAX_VARCHAR);
                 return Optional.of(varcharColumnMapping(
                         length < VarcharType.MAX_LENGTH
```

Comment:
Do you see any scenario where `length` will go beyond `65535`?

Reply:
Now it defaults to `VarcharType.MAX_LENGTH` so not an issue.